File: src/1945. Sum of Digits of String After Convert/1945.py
class Solution:
    def getLucky(self, s: str, k: int) -> int:
        digits = "".join(str(ord(ch)-ord("a")+1) for ch in s)
        for i in range(k):
            if len(digits) == 1:
                break
            total=sum(int(ch) for ch in digits)
            digits = str(total)
        return int(digits)

if __name__ == "__main__":
    s = "leetcode"
    k = 2
    print(Solution().getLucky(s, k))

File: src/1945. Sum of Digits of String After Convert/1945.go
func getLucky(s string, k int) int {
	t := []byte{}
	for _, c := range s {
		t = append(t, strconv.Itoa(int(c-'a'+1))...)
	}
	digits := string(t)
	for i := 1; i <= k && len(digits) > 1; i++ {
		sum := 0
		for _, c := range digits {
			sum += int(c - '0')
		}
		digits = strconv.Itoa(sum)
	}
	ans, _ := strconv.Atoi(digits)
	return ans
}

